qiskit_experiments/curve_analysis/composite_curve_analysis.py

```diff
@@ -0,0 +1,452 @@
+        self,
+        experiment_data: ExperimentData,
+    ):
+        """Initialize curve analysis with experiment data.
+
+        This method is called ahead of other processing.
+
+        Args:
+            experiment_data: Experiment data to analyze.
+        """
+        for analysis in self._analyses:
+            analysis._initialize(experiment_data)
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Default analysis options."""
+        options = BaseCurveAnalysis._default_options()
+        options.curve_drawer = MplCurveDrawer()
+        options.plot_raw_data = False
+        options.plot = True
+        options.return_fit_parameters = True
+        options.return_data_points = False
+
+        # Set automatic validator for particular option values
+        options.set_validator(field="curve_drawer", validator_value=BaseCurveDrawer)
+
+        return options
+
+    def set_options(self, **fields):
+        for field in fields:
+            if not hasattr(self.options, field):
+                warnings.warn(
+                    f"Specified option {field} doesn't exist in this analysis instance. "
+                    f"Note that {self.__class__.__name__} is a composite curve analysis instance, "
+                    "which consists of multiple child curve analyses. "
+                    "This options may exist in each analysis instance. "
+                    "Please try setting options to child analyses through '.analyses()'.",
+                    UserWarning,
+                )
+        super().set_options(**fields)
+
+    def _run_analysis(
+        self,
+        experiment_data: ExperimentData,
+    ) -> Tuple[List[AnalysisResultData], List["matplotlib.figure.Figure"]]:
+
+        analysis_results = []
+
+        self.drawer.initialize_canvas()
+
+        # Initialize canvas
+        if self.options.plot:
+            self.drawer.initialize_canvas()
+
+        fit_dataset = {}
+        for analysis in self._analyses:
+            metadata = analysis.options.extra.copy()
+            metadata["group"] = analysis.name
+
+            processed_data = analysis._run_data_processing(
+                raw_data=experiment_data.data(),
+                models=analysis.models,
+            )
+
+            if self.options.plot and self.options.plot_raw_data:
+                for model in analysis.models:
+                    sub_data = processed_data.get_subset_of(model._name)
+                    self.drawer.draw_raw_data(
+                        x_data=sub_data.x,
+                        y_data=sub_data.y,
+                        name=model._name + f"_{analysis.name}",
+                    )
+
+            # Format data
+            formatted_data = analysis._format_data(processed_data)
+            if self.options.plot:
+                for model in analysis.models:
+                    sub_data = formatted_data.get_subset_of(model._name)
+                    self.drawer.draw_formatted_data(
```

Comment:
I have not looked at the drawer code. Does it draw everything on a single plot? So the different groups will be drawn on the same plot?

Reply:
Yes, it instantiates a drawer on the composite instance, and uses that drawer to run visualization code on sub analysis. In this way, the composite analysis draws all curves on the same plot.